### PR TITLE
cereal: add new versions

### PR DESCRIPTION
**Pull request by s-sajid-ali**
Add cereal versions `1.3.1` and `1.3.2`.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cereal/package.py [hunk 1]
```diff
@@ -28,13 +30,14 @@ class Cereal(CMakePackage):
...
-    patch("Boost2.patch", when="@1.2.2:")
+    patch("Boost.patch", when="@:1.3.0")
+    patch("Boost2.patch", when="@1.2.2:1.3.0")
     patch("pointers.patch")
     # fixed in HEAD but not released yet
-    patch("LockGuard-default-ctor.patch", when="@:1.3")
+    patch("LockGuard-default-ctor.patch", when="@:1.3.0")
 
-    depends_on('cmake@2.6.2:', type='build')
+    depends_on('cmake@2.6.2:', when="@:1.3.0", type='build')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cereal/package.py]:
```suggestion
    depends_on('cmake@2.6.2:', type='build')
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request

**Comment by s-sajid-ali** [comment]
The failing job in the gitlab-ci is stage15 of the ` radiuss-pr-build `, `(specs) lbann/l3nnna7 0.102 gcc@7.5.0 linux-ubuntu18.04-x86_64 RADIUSS`. It fails with: 
```
1 error found in build log:
     43    -- Looking for pthread_create in pthreads - not found
     44    -- Looking for pthread_create in pthread
     45    -- Looking for pthread_create in pthread - found
     46    -- Found Threads: TRUE
     47    -- Found Clara: /home/software/radiuss/__spack_path_placeholder__/__
           spack_path_placeholder__/__spack_path_placeholder__/__spack_path_pla
           ceholder__/__spack_path_placeholder__/__spack_path_placeholder__/__s
           pack_path_placeholder__/__spack_path_placeholder__/__spack_path_plac
           eholder__/__spack_path_placeholder__/__spack_path_placeholder__/__sp
           ack_path_placeholder__/__spack_path_placeholder__/__spack_path_place
           holder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa
           ck_path_placeholder__/__spack_path_placeholder__/__s/linux-ubuntu18.
           04-x86_64/gcc-7.5.0/clara-1.1.5-5broixoohwd7evl7u2ue7mhi76la4xus/inc
           lude
     48    -- Could NOT find CEREAL (missing: CEREAL_DIR)
  >> 49    CMake Error at CMakeLists.txt:200 (find_package):
     50      Could not find a package configuration file provided by "CEREAL" w
           ith any
     51      of the following names:
     52    
     53        CEREALConfig.cmake
     54        cereal-config.cmake
     55    
See build log for details:
  /tmp/root/spack-stage/spack-stage-lbann-0.102-l3nnna7f2f2pan6lcqq3rg4phrgikmu6/spack-build-out.txt
```
but I see that `CEREAL_DIR` is passed to cmake as `-DCEREAL_DIR=/home/software/radiuss/[padded-to-512-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/cereal-1.3.2-3dxu3h3hkqmfkemqwbir7yk3u6sfreuh`. I don't understand why `lbann` doesn't seem to find it.

**Review by alalazo — DISMISSED** [review]

**Comment by s-sajid-ali** [comment]
Pinging lbann maintainer regarding the build failure of lbann with this PR @bvanessen.

**Comment by bvanessen** [comment]
@s-sajid-ali Let me take a look at it.

**Comment by s-sajid-ali** [comment]
Could it be the case that one needs to set the package name in lbann as lowercase `cereal` instead of uppercase as is currently done?

On my laptop I see : 
```
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/l/c/cereal (develop)> pwd                                                                                                                                                                        (py3)
/home/sajid/packages/spack/opt/spack/linux-ubuntu20.04-zen2/clang-13.0.1/cereal-1.3.2-f7327zon6djtgyz2kggfspy42m2mu6co/lib/cmake/cereal
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/l/c/cereal (develop)> ls                                                                                                                                                                         (py3)
cerealConfig.cmake  cerealConfigVersion.cmake  cerealTargets.cmake
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/l/c/cereal (develop)>   
```

**Comment by s-sajid-ali** [comment]
Could it be the case that one needs to set the package name in lbann as lowercase `cereal` instead of uppercase as is currently done?

On my laptop I see the following for `cereal@1.3.2`: 
```
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/l/c/cereal (develop)> pwd                                                                                                                                                                        (py3)
/home/sajid/packages/spack/opt/spack/linux-ubuntu20.04-zen2/clang-13.0.1/cereal-1.3.2-f7327zon6djtgyz2kggfspy42m2mu6co/lib/cmake/cereal
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/l/c/cereal (develop)> ls                                                                                                                                                                         (py3)
cerealConfig.cmake  cerealConfigVersion.cmake  cerealTargets.cmake
```
but for `cereal@1.3.0`, I have: 
```
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/s/c/cereal (develop)> pwd
/home/sajid/packages/spack/opt/spack/linux-ubuntu20.04-zen2/clang-13.0.1/cereal-1.3.0-yfqggk3z26uikozun6yrdqvewfmcsyhq/share/cmake/cereal
sajid@LAPTOP-CDJT2P3R ~/p/s/o/s/l/c/c/s/c/cereal (develop)> ls
cereal-config.cmake
```

The [cmake docs for `find_package`](https://cmake.org/cmake/help/latest/command/find_package.html#search-modes) state that : `In this mode (Config mode), CMake searches for a file called <lowercasePackageName>-config.cmake or <PackageName>Config.cmake. It will also look for <lowercasePackageName>-config-version.cmake or <PackageName>ConfigVersion.cmake `. This is probably why cmake finds `cereal@1.3.0` but not `cereal@1.3.2` with `find_package(CEREAL)`.

**Comment by s-sajid-ali** [comment]
Ping @alalazo: would it possible to handle the `lbann` issue of converting the package name from uppercase to lowercase elsewhere (a separate PR) ?

**Comment by alalazo** [comment]
@s-sajid-ali CI is failing legitimately, so it should be fixed here. Otherwise we'd introduce broken builds, which is what we are trying to avoid with Gitlab.

**Comment by s-sajid-ali** [comment]
Sounds good. Added a conflict until `lbann` fixes the `find_package` to use `cereal@1.3.1:`.

**Comment by alalazo** [comment]
I'll give some time to @bvanessen to comment on this approach, but LGTM from a packaging perspective

**Review by alalazo — APPROVED** [review]

**Comment by bvanessen** [comment]
@s-sajid-ali @alalazo We just pushed a fix into LBANN's develop branch that should address this issue.

**Pull request merged**